### ISSUE TITLE
AL02 incorrectly flags T-SQL alias syntax when using DISTINCT TOP

### ISSUE THREAD

**Issue by fank** [quarylabs/sqruff#1750]
## Description

The AL02 rule incorrectly flags valid T-SQL column alias syntax (`ColumnAlias = expression`) as implicit aliasing when used with `SELECT DISTINCT TOP`. This is a false positive that only occurs when both `DISTINCT` and `TOP` are present.

## SQL Example

```sql
-- This incorrectly triggers AL02
SELECT DISTINCT TOP 20
    JiraIssueID = JiraIssue.i_jira_id,
    JiraIssueKey = JiraIssue.s_key
FROM JiraIssue
```

Error:
```
L:   2 | P:   2 | AL02 | Implicit/explicit aliasing of columns. [aliasing.column]
```

## Expected Behavior

T-SQL's `alias = expression` syntax is valid and should not trigger AL02. This syntax works correctly in other cases:

```sql
-- These work fine (no AL02 error)
SELECT ColumnAlias = expression FROM table1;  -- ✓ Works
SELECT DISTINCT ColumnAlias = expression FROM table1;  -- ✓ Works
SELECT TOP 20 ColumnAlias = expression FROM table1;  -- ✗ Incorrectly triggers AL02
SELECT DISTINCT TOP 20 ColumnAlias = expression FROM table1;  -- ✗ Incorrectly triggers AL02
```

## Root Cause Analysis

After extensive debugging, I found that the parser is misinterpreting the structure when `DISTINCT TOP` is present:

1. The parser correctly identifies `DISTINCT` as a SelectClauseModifier
2. But then `TOP 20` is incorrectly parsed as part of the SelectClauseElement (as an Expression)
3. `ColumnAlias` is then parsed as an AliasExpression for that expression
4. This triggers AL02 because it appears to be implicit aliasing: `expression alias` pattern

Debug output shows the incorrect parsing:
```
Parent SelectClauseElement segments:
  - Expression: "TOP 20"  // ← This should be in the modifier!
  - Newline: "\n"
  - Whitespace: "    "
  - AliasExpression: "ColumnAlias"
```

The `= expression` part is parsed as a separate element entirely.

### 1. Grammar Fix for SelectClauseModifierSegment

I modified the T-SQL dialect to properly handle `DISTINCT TOP` combinations:

```rust
// In crates/lib-dialects/src/tsql.rs
dialect.replace_grammar(
    "SelectClauseModifierSegment",
    one_of(vec_of_erased![
        // DISTINCT TOP combination (must come before just DISTINCT)
        Sequence::new(vec_of_erased![
            Ref::keyword("DISTINCT"),
            Ref::keyword("TOP"),
            one_of(vec_of_erased![
                Ref::new("NumericLiteralSegment"),
                Ref::new("TsqlVariableSegment"),
                // ... (full TOP clause grammar)
            ]),
            Ref::keyword("PERCENT").optional(),
            Ref::keyword("WITH").optional(),
            Ref::keyword("TIES").optional()
        ]),
        // ... other patterns ...
    ])
);
```

### 2. AL02 Rule Modification

I also modified the AL02 rule to check for T-SQL's `alias = expression` syntax:

```rust
// In crates/lib/src/rules/aliasing/al02.rs
fn eval(&self, context: &RuleContext) -> Vec<LintResult> {
    // Check if this AliasExpression is part of "alias = expression" syntax
    if let Some(parent) = context.parent_stack.last() {
        if parent.get_type() == SyntaxKind::SelectClauseElement {
            let parent_segments = parent.segments();
            
            // Find this AliasExpression in the parent's children
            if let Some(pos) = parent_segments.iter().position(|s| s == &context.segment) {
                // Check if the next non-whitespace segment is "="
                for i in (pos + 1)..parent_segments.len() {
                    let seg = &parent_segments[i];
                    if !seg.is_whitespace() && !seg.is_meta() {
                        if seg.raw() == "=" {
                            // This is T-SQL "alias = expression" syntax
                            return Vec::new();
                        }
                        break;
                    }
                }
            }
        }
    }
    // ... rest of the rule ...
}
```

## Issue Status

Despite these fixes, the issue persists. The grammar changes don't seem to take effect - the parser still incorrectly includes `TOP 20` in the SelectClauseElement instead of the SelectClauseModifier. This suggests either:

1. The grammar replacement mechanism isn't working as expected
2. There's a parser precedence issue that needs deeper investigation
3. The SelectClause parsing logic needs additional changes

## Environment

- sqruff version: 0.28.0
- Rust version: 1.88.0
- Dialect: tsql

## Reproduction Steps

1. Create a file with the SQL example above
2. Run: `sqruff lint --dialect tsql file.sql`
3. Observe the AL02 error on line 2

## Workaround

Until this is fixed, users can:
1. Use `-- noqa: AL02` to suppress the error
2. Use alternative syntax: `JiraIssue.i_jira_id AS JiraIssueID`
3. Exclude AL02 for T-SQL files in configuration

## Additional Context

The T-SQL dialect correctly defines the `alias = expression` pattern in SelectClauseElementSegment, and it works in most cases. The issue is specific to when `TOP` appears after `DISTINCT` or `ALL`, causing the parser to misinterpret the structure.
